src/firestore/index.ts

```diff
@@ -96,13 +110,29 @@ export {
  *   service if no app is provided or the `Firestore` service associated with the
  *   provided app.
  */
-export function getFirestore(app?: App): Firestore {
-  if (typeof app === 'undefined') {
-    app = getApp();
-  }
-
+export function getFirestore(app: App): Firestore;
+/**
+ * @param databaseId
+ * @internal
+ */
+export function getFirestore(databaseId: string): Firestore;
+/**
+ * @param app
+ * @param databaseId
+ * @internal
+ */
+export function getFirestore(app: App, databaseId: string): Firestore;
+export function getFirestore(
+  appOrDatabaseId?: App | string,
+  optionalDatabaseId?: string
+): Firestore {
+  const app: App = typeof appOrDatabaseId === 'object' ? appOrDatabaseId : getApp();
+  const databaseId =
+    typeof appOrDatabaseId === 'string'
```

Comment:
Do we want to check if `appOrDatabaseId` is a non-empty string? We a validator util if this is needed `validator.isNonEmptyString()`.

Reply:
Good point. I think we should consider an empty string equivalent to the default database id.